tests/impl.cpp

```diff
@@ -3000,6 +3011,43 @@ bool test_mm_crc32_u64(uint64_t crc, uint64_t v)
     return true;
 }
 
+
+bool test_mm_ceil_ps(const float *_a)
+{
+    float dx = ceilf(_a[0]);
+    float dy = ceilf(_a[1]);
+    float dz = ceilf(_a[2]);
+    float dw = ceilf(_a[3]);
+
+    __m128 a = test_mm_load_ps(_a);
+    __m128 c = _mm_ceil_ps(a);
+    return validateFloatEpsilon(c, dx, dy, dz, dw, 300.0f);
```

Comment:
I wonder what epsilon value should be set.
Wouldn't `300.0f` be too large?
@jserv any idea?

Reply:
At present, `validateFloatEpsilon` appears only once. `300.0f` is definitely too rough. We can calculate via the definition of ceil operation.

Reply:
For floating point numbers, a == b is risky due to the way they are encoded. a == (b + epsilon) is safer.
Functions such as cos, sin, tan, etc, are implemented using Taylor expansion, hence some shortcut might give different values. The same trouble goes with fused multiple add functions, which are often less precise than separate multiply plus add.

The impl.cpp uses this as input vectors :
`        for (uint32_t i = 0; i < MAX_TEST_VALUE; i++) {
            mTestFloats[i] = ranf(-100000, 100000);
            mTestInts[i] = (int32_t) ranf(-100000, 100000);
        }`

In my opinion, floating point numbers are great for describing small values, and not so much for large ones. I tend to manipulate small values and compute the L2 error, for which something between 10^-5 and 10^-7 is a good result.

Without changing the test vector, I lowered the epsilon to 5.0f.

I did not implement any test for _mm_blendv_ps yet, but the function is used for _mm_round_ps in AArch32 mode.